src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -363,4 +412,79 @@ protected int doHashCode() {
     public String getWriteableName() {
         return NAME;
     }
+
+    /**
+     * Builder for KNNQueryBuilder class to create KNNQueryBuilder, which can be used to build k-NN query or r-NN query.
+     * */
+    public static class Builder {
+        private final String fieldName;
+        private final float[] vector;
+        private Integer k = null;
+        private Float distance;
+        private QueryBuilder filter;
+        private boolean ignoreUnmapped;
+        private String queryName;
+        private float boost = AbstractQueryBuilder.DEFAULT_BOOST;
+
+        public Builder(String fieldName, float[] vector) {
+            if (Strings.isNullOrEmpty(fieldName)) {
+                throw new IllegalArgumentException("[" + NAME + "] requires fieldName");
+            }
+            if (vector == null) {
+                throw new IllegalArgumentException("[" + NAME + "] requires query vector");
+            }
+            if (vector.length == 0) {
+                throw new IllegalArgumentException("[" + NAME + "] query vector is empty");
+            }
+            this.fieldName = fieldName;
+            this.vector = vector;
+        }
+
+        public Builder k(Integer k) {
+            this.k = k;
+            return this;
+        }
+
+        public Builder distance(Float distance) {
+            this.distance = distance;
+            return this;
+        }
+
+        public Builder filter(QueryBuilder filter) {
+            this.filter = filter;
+            return this;
+        }
+
+        public Builder ignoreUnmapped(boolean ignoreUnmapped) {
+            this.ignoreUnmapped = ignoreUnmapped;
+            return this;
+        }
+
+        public Builder queryName(String queryName) {
+            this.queryName = queryName;
+            return this;
+        }
+
+        public Builder boost(float boost) {
+            this.boost = boost;
+            return this;
+        }
+
+        public KNNQueryBuilder build() {
+            if (k != null && distance != null || k == null && distance == null) {
```

Comment:
Split this into two condition eg: if both are null and both are non-null

Reply:
Updated